libs/animation/animation.ts

```diff
@@ -407,6 +436,120 @@ namespace animation {
         }
     }
 
+    function isClearPath(interval: number, target: Sprite, x0: number, y0: number, x1: number, y1: number): boolean {
+        const tm = game.currentScene().tileMap;
+        if (tm && tm.enabled ) {
+            const dx = x1 - x0;
+            const dy = y1 - y0;
+            const ms = new MovingSprite(
+                target,
+                Fx8(dx / interval),
+                Fx8(dy / interval),
+                Fx8(dx),
+                Fx8(dy),
+                Fx8(dx),
+                Fx8(dy)
+            );
+            return !hasCollision(ms, tm, dx, dy);
+
+        }
+        return true;
+    }
+
+    // This is using a modified version of tilemapCollisions()
+    function hasCollision(movingSprite: MovingSprite, tm: tiles.TileMap, dx: number, dy: number): boolean {
```

Comment:
Should this be in the physics engine? I don't like duplicating this much code here. What changes are there?

Reply:
- I cut out the collision & overlapped tiles arrays
- Instead of checking if xDiff or yDiff is bigger than zero to check, I'm checking velocity instead
- There was also the bounce logic, clipping logic, and other movement things that I took out